Migrate Docs from GitBook (#112)
mattermost/mattermost-plugin-splunk: README.md

**Maintainer:** [@dbejanishvili](https://github.com/dbejanishvili)
**Co-Maintainers:** [@bakurits](https://github.com/bakurits) [@Gvantsats](https://github.com/Gvantsats)

A Splunk integration for Mattermost which enables users to get logs and alerts from Splunk server. 

## Admin guide

> ### Marketplace installation

Author: ```suggestion
### Installation

#### Marketplace installation
```

**Maintainer:** [@dbejanishvili](https://github.com/dbejanishvili)
**Co-Maintainers:** [@bakurits](https://github.com/bakurits) [@Gvantsats](https://github.com/Gvantsats)

A Splunk integration for Mattermost which enables users to get logs and alerts from Splunk server. 

## Admin guide

### Installation

#### Marketplace installation
1. In Mattermost, go to **Main Menu > Plugin Marketplace**.
2. Search for "Splunk" or find the plugin from the list.
3. Select **Install**.
4. When the plugin has downloaded and been installed, select **Configure**.

> ### Manual installation

Author: ```suggestion
#### Manual installation
```

The `SPLUNK_PASSWORD` environment variable is set to `SplunkPass`, as defined in the `docker-compose.yml` file. You can login with these credentials:

Username: `admin`

Password: `SplunkPass`

If you want to modify the default Alert hostname, you can do so editing the `default.yml` file and replace `<MY_ALERT_HOSTNAME>` with your valid hostname (ex: `https://myhost.ngrok.io`).

### Build and deploy
To build your plugin use `make`, you can use `MM_DEBUG=1` as an envvar to generate a debug version of the plugin, including an unminified version of the Javascript webapp.

Use make `check-style` to check the style, use `make dist` and make deploy to build and deploy the application.
`make` will produce a single plugin file (with support for multiple architectures) for upload to your Mattermost server:

`dist/com.example.my-plugin.tar.gz`

Alternatively you can deploy a plugin automatically to your server, but it requires login credentials:

```
export MM_SERVICESETTINGS_SITEURL=http://localhost:8065
export MM_ADMIN_USERNAME=admin
export MM_ADMIN_PASSWORD=password
make deploy
```

> or configuration of a [personal access token](https://docs.mattermost.com/developer/personal-access-tokens.html):

Author: ```suggestion
or configuration of a [personal access token](https://developers.mattermost.com/integrate/reference/personal-access-token/):
```
